pkg/tfbridge/schema.go

```diff
@@ -1103,14 +1103,27 @@ func getDefaultValue(tfs shim.Schema, ps *SchemaInfo) interface{} {
 	if dv, _ := tfs.DefaultValue(); dv != nil {
 		return dv
 	}
-	if ps != nil {
-		return ps.Default
-	}
+	// TODO: We should inspect SchemaInfo.Default for the default value as well
+	// if ps != nil {
+	// 	return ps.Default
+	// }
```

Comment:
Curious about this one. Why was this commented out?

Reply:
`SchemaInfo.Default` is a `*DefaultInfo`. Not the actual default value. I'm sure there's a way to turn a `DefaultInfo` into the value given the info on it:
```
// DefaultInfo lets fields get default values at runtime, before they are even passed to Terraform.
type DefaultInfo struct {
	// AutoNamed is true if this default represents an autogenerated name.
	AutoNamed bool
	// Config uses a configuration variable from this package as the default value.
	Config string
	// From applies a transformation from other resource properties.
	From func(res *PulumiResource) (interface{}, error)
	// Value injects a raw literal value as the default.
	Value interface{}
	// EnvVars to use for defaults. If none of these variables have values at runtime, the value of `Value` (if any)
	// will be used as the default.
	EnvVars []string
}
```

But that's a bigger change than just trying to fix secrets and integers. So comment out and TODO.

Reply:
> `SchemaInfo.Default` is a `*DefaultInfo`. Not the actual default value. 

:facepalm: